cache-config/tm-health-client/tm-health-client.service

```diff
@@ -18,13 +18,14 @@
 #
 [Unit]
 Description=Traffic monitor health service for Trafficserver
-After=trafficserver
+After=trafficserver.service
+Requires=trafficserver.service
 
 [Service]
 ExecStart=/usr/bin/tm-health-client -vv
-RequiredBy=trafficserver
 Restart=always
-ExecStop=/bin/kill -s $MAINPID
+ExecStop=/bin/kill -p $MAINPID
```

Comment:
I don't think this will actually kill the process. From [the man page for `kill(1)`](https://man7.org/linux/man-pages/man1/kill.1.html#OPTIONS):
>        -p, --pid
>           Only print the process ID (PID) of the named processes, do not send any signals.

Reply:
I think instead `-s` should still be used, but it should come with a numeric or macro name for the signal to be sent. Usually when you bother to specify (and actually want program termination) you'd send `-s 9` or `-s SIGKILL`. Using no options at all sends `SIGTERM`, I think.

Reply:
You are right. I saw that pid number in the log file and wondered why. Somehow the program gets killed anyway from my testing. But that should be fixed.

I also think we should add the `RestartSec=5` so it doesn't fail on a loop too quick.

```
[root@cdn1cdedge0004 system]# systemctl stop trafficserver
[root@cdn1cdedge0004 system]# 
[root@cdn1cdedge0004 system]# 
[root@cdn1cdedge0004 system]# systemctl status tm-health-client
● tm-health-client.service - Traffic monitor health service for Trafficserver
   Loaded: loaded (/usr/lib/systemd/system/tm-health-client.service; enabled; vendor preset: disabled)
   Active: inactive (dead) since Wed 2021-09-08 18:57:30 UTC; 3s ago
  Process: 5865 ExecStop=/bin/kill -p $MAINPID (code=exited, status=0/SUCCESS)
  Process: 5850 ExecStart=/usr/bin/tm-health-client -vv (code=killed, signal=TERM)
 Main PID: 5850 (code=killed, signal=TERM)

Sep 08 18:57:15 cdn1cdedge0004.coxlab.net systemd[1]: Started Traffic monitor health service for Trafficserver.
Sep 08 18:57:15 cdn1cdedge0004.coxlab.net systemd[1]: Starting Traffic monitor health service for Trafficserver...
Sep 08 18:57:30 cdn1cdedge0004.coxlab.net systemd[1]: Stopping Traffic monitor health service for Trafficserver...
Sep 08 18:57:30 cdn1cdedge0004.coxlab.net kill[5865]: 5850
Sep 08 18:57:30 cdn1cdedge0004.coxlab.net systemd[1]: Stopped Traffic monitor health service for Trafficserver.
[root@cdn1cdedge0004 system]# systemctl start trafficserver
[root@cdn1cdedge0004 system]# systemctl status tm-health-client
● tm-health-client.service - Traffic monitor health service for Trafficserver
   Loaded: loaded (/usr/lib/systemd/system/tm-health-client.service; enabled; vendor preset: disabled)
   Active: active (running) since Wed 2021-09-08 18:57:37 UTC; 1s ago
  Process: 5865 ExecStop=/bin/kill -p $MAINPID (code=exited, status=0/SUCCESS)
 Main PID: 5889 (tm-health-clien)
   CGroup: /system.slice/tm-health-client.service
           └─5889 /usr/bin/tm-health-client -vv

Sep 08 18:57:37 cdn1cdedge0004.coxlab.net systemd[1]: Started Traffic monitor health service for Trafficserver.
Sep 08 18:57:37 cdn1cdedge0004.coxlab.net systemd[1]: Starting Traffic monitor health service for Trafficserver...
```

Reply:
According to documentation, ExecStop is optional and the program will be stopped using KillMode=control-group (by default).

https://www.freedesktop.org/software/systemd/man/systemd.service.html#ExecStop=

You can also read about the order in which the process is tried to be terminated here:
https://www.freedesktop.org/software/systemd/man/systemd.kill.html#KillMode=